src/backend/utils/storage.js

```diff
@@ -54,19 +78,34 @@ module.exports = {
         post.guid
       )
       // sort set by published date as scores
-      .zadd(POSTS, post.published.getTime(), post.guid)
+      .zadd(POSTS, post.published.getTime(), key)
       .exec();
   },
 
   /**
-   * Gets an array of guids from redis
+   * Returns an array of guids from redis
    * @param from lower index
    * @param to higher index, it needs -1 because redis includes the element at this index in the returned array
    * @return Array of guids
    */
-  getPosts: (from, to) => redis.zrevrange(POSTS, from, to - 1),
+  getPosts: async (from, to) => {
+    const keys = await redis.zrevrange(POSTS, from, to - 1);
+
+    /**
+     * 'zrevrange returns an array of encrypted hashed guids.
```

Comment:
NOTE: they aren't encrypted, only hashed (sha256) and encoded (base64).

Reply:
Fixed comment